analytics/views.py

```diff
@@ -243,7 +243,7 @@ def get_chart_data(request: HttpRequest, user_profile: UserProfile, chart_name:
         if end is None:
             end = max(last_successful_fill(stat.property) or
                       datetime.min.replace(tzinfo=timezone_utc) for stat in stats)
-        if end is None or start > end:
+        if end is None or start > end:  # type: ignore # ->warn-unreachable
```

Comment:
I'm actually curious about what's going on with double checking `if end is None`.  I know `last_successful_fill` might return `None`, but is it possible nothing will be returned from the comprehension of `stats`?

Reply:
There’s indeed no way for `end` to still be `None` here, so let’s remove the check. Even if `last_successful_fill` returns `None`, `datetime.min.replace` can’t. And it’s not an empty sequence thing either; `max([])` raises `ValueError`.